modal/_function_utils.py

```diff
@@ -297,3 +298,25 @@ def get_tag(self):
 
     def is_nullary(self):
         return all(param.default is not param.empty for param in self.signature.parameters.values())
+
+
+def get_referred_objects(f: Callable) -> List[Object]:
+    """Takes a function and returns any Modal Objects in global scope that it refers to.
+
+    TODO: this does not yet support Object contained by another object,
+    e.g. a list of Objects in global scope.
+    """
+    from .cls import Cls
+    from .functions import Function
+
+    ret: List[Object] = []
+    for obj in inspect.getclosurevars(f).globals.values():
+        if isinstance(obj, (Function, Cls)):
+            # These are always attached to stubs, so we shouldn't do anything
+            pass
+        elif isinstance(obj, Object):
+            ret.append(obj)
+        elif inspect.isfunction(obj):
+            ret += get_referred_objects(obj)
```

Comment:
can this infinite loop with two functions that call each other?

Reply:
haha shit you're right – let me fix (and add a test)

Reply:
Fixed this by not using recursion, and keeping track of all objects seen so far.